### PR TITLE
Alias paymentID to orderID experimentally

### PR DESCRIPTION
**Pull request by jshawl**
### Description

This PR populates `paymentID` with the value of `orderID` if `paymentID` is falsey _and_ if an experiment to enable this functionality is set to true. 

### Why are we making these changes?

Populating `paymentID` with the value of `orderID` will allow the Braintree SDK to make changes to `tokenizePayment` without asking merchants to upgrade to the latest version of the Braintree SDK.

### Reproduction Steps (if applicable)

Currently, this is only reproducible for Braintree merchants that have been added to an allow-list. In the meantime, you can run `npm run vitest onApprove` to run the unit tests.

❤️  Thank you!

### CODE REVIEW

File: src/props/onApprove.js [hunk 1]
```diff
@@ -392,7 +393,9 @@ export function getOnApproveOrder({ intent, onApprove = getDefaultOnApproveOrder
             return getSupplementalOrderInfo(orderID).then(supplementalData => {
                 billingToken = billingToken || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.billingToken);
                 paymentID = paymentID || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.paymentId);
-
+                if(experiments.btSdkOrdersV2Migration && !paymentID){
+                    paymentID = orderID;
+                }
```
Comment (jshawl) [inline src/props/onApprove.js:398]: I don't think we have codecov support for `vitest` yet but this is covered here https://github.com/paypal/paypal-smart-payment-buttons/pull/642/files#diff-4e744334f2eab852b4cb56a6fea9f9d810821aa9eca431e3341068bf90bf5b31R47

**Review by nbierdeman — APPROVED** [review]
Nice work! 🥇

File: src/props/onApprove.js [hunk 2]
```diff
@@ -392,7 +393,9 @@ export function getOnApproveOrder({ intent, onApprove = getDefaultOnApproveOrder
             return getSupplementalOrderInfo(orderID).then(supplementalData => {
                 billingToken = billingToken || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.billingToken);
                 paymentID = paymentID || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.paymentId);
-
+                if(experiments.btSdkOrdersV2Migration && !paymentID){
+                    paymentID = orderID;
```
Comment (jshawl) [inline src/props/onApprove.js:397]: remove `EC-` from `orderID`

File: src/props/onApprove.js [hunk 3]
```diff
@@ -392,7 +393,12 @@ export function getOnApproveOrder({ intent, onApprove = getDefaultOnApproveOrder
             return getSupplementalOrderInfo(orderID).then(supplementalData => {
                 billingToken = billingToken || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.billingToken);
                 paymentID = paymentID || (supplementalData && supplementalData.checkoutSession && supplementalData.checkoutSession.cart && supplementalData.checkoutSession.cart.paymentId);
-
+                if(experiments.btSdkOrdersV2Migration && !paymentID){
+                    // The Braintree SDK prefixes the orderID with "EC-" to support
+                    // backwards compatibility. Before invoking onApprove, we need
+                    // to remove that prefix.
```
Comment (nbierdeman) [inline src/props/onApprove.js:399]: I'm thinking we either remove this comment or add something similar to the equivalent block in getOnShippingChange(). All or nothing. 😁

**Review by nbierdeman — APPROVED** [review]
💯 🚢

File: src/props/onApprove.js (hunk 3 above)
Comment (jshawl) [reply src/props/onApprove.js:399]: oo good catch :pray:

**Pull request merged**